### ISSUE TITLE
TLS WI-Fi authentication

### ISSUE THREAD

**Issue by genehunter29009** [Dynatrace/ufo-esp32#28]
I have a question about TLS authentication. I flashed the firmware onto an esp32 to test without blowing up the UFO. I entered a logon name, password and Cert, however I am getting an error message that private key not configured. What do i need to do to get this running?

I see the following options. I do not understand what i need to put in here. the error message indicates I need a private key. 

SSID we know that
Enterprise WPA2 Client Certificate(chain)
Enterprise WPA2 Client Certificate Key ( without password)
Hostname we have that
Enterprise WPA2 Trusted Certificate Authority(optional)

[0;32mI (98415) wpa: EAP-TLS: Private key not configured[0m
E (98415) wpa: Method private structure allocated failure

[0;32mI (98450) wpa: SSL: Need 3315 bytes more input data[0m
[0;32mI (98460) wpa: SSL: Need 2313 bytes more input data[0m
[0;32mI (98475) wpa: SSL: Need 1311 bytes more input data[0m
[0;32mI (98495) wpa: SSL: Need 309 bytes more input data[0m
I (98975) wpa: >>>>>wpa2 FAILED

**Comment by helmut-s** [comment]
The certificate you provided is a client cert including the private key? If yes what format does it have?

**Comment by genehunter29009** [comment]
I am testing with the Cert from my laptop and using the same Ad credentials i use on my laptop. I exported the Cert however I am not allowed to export the private key. I dont know a lot about enterprise Wi-Fi. do I need the private key as the message indicates and where would I get that? the networking team? also what format does it need to be coded? I believe i used base 64 or something like that.

**Comment by helmut-s** [comment]
yes the key is needed and base 64 would be the correct format. The webui shows an example (cert and key section need to be present). I would ask the IT department for a matching file

**Comment by genehunter29009** [comment]
Thank you. I will let you know how it worked out.
